ui/src/containers/SampleGridTableContainer.jsx

```diff
@@ -965,6 +1053,135 @@ class SampleGridTableContainer extends React.Component {
     return tableCell;
   }
 
+  renderPuck(puck) {
+    if (
+      Number(this.props.filterOptions.puckFilter) === Number(puck.name) ||
+      this.props.filterOptions.puckFilter.toLowerCase() === ''
+    ) {
+      const puckMenuID = 'samples-grid-table-context-menu-puck';
+      return (
+        <th
+          key={`${puck.name}-th`}
+          className="sample-items-table-col-header-th"
+        >
+          <span
+            style={{
+              marginLeft: '5px',
+              marginTop: '6px',
+              float: 'left',
+            }}
+          >
+            Puck {puck.name}
+          </span>
+          <span
+            style={{
+              marginTop: '15px',
+              marginRight: '2px',
+            }}
+          >
+            {this.itemsControls(1, Number(puck.name))}
+          </span>
+          <span
+            title="Puck Options"
+            className="samples-grid-table-context-menu-icon"
+            onClick={(e) => {
+              this.displayPuckCellContextMenu(
+                e,
+                puckMenuID,
+                Number(puck.name),
+                1, //idxth + 1,
+              );
+            }}
+          >
+            <BiMenu size="1.5em" />
+          </span>
+        </th>
+      );
+    }
+    return null;
+  }
+
+  getSampleTableSingleCell(colsm) {
+    const scContent = this.props.sampleChanger.contents;
+    const tableCell = [];
+
+    // Do not try to render is SC is empty
+    if (typeof scContent.children === 'undefined') {
+      return null;
+    }
+
+    // Do not try to render is samplesList is empty
+    // only Puck header would be drawn
+    if (Object.keys(this.props.sampleList).length === 0) {
+      return null;
+    }
+
+    scContent.children.map((puck) => {
```

Comment:
I think this could be written as:

```
return scContent.children.map(...);
```

... instead of initialising and pushing to the `tableCell` array.